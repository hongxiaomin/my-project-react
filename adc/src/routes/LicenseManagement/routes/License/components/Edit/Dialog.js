import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { Form, Input, Select, Icon, Button, Modal } from 'antd'
import { trim } from 'lodash'
import { getRole, showMessage } from 'Utils'

const Option = Select.Option
const FormItem = Form.Item
export class DialogFilter extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      loading: false,
      passwordInputType: 'password',
    }
    this.handleChange = this.handleChange.bind(this)
  }

  mouseDownEye = () => {
    console.log('this.mouseDown')
    this.setState({
      passwordInputType: 'text',
    })
  }

  mouseUpEye = () => {
    this.setState({
      passwordInputType: 'password',
    })
  }

  handleChange(e) {
    e.preventDefault()
    this.props.form.validateFields((err, objValues) => {
      if (!err) {
        this.setState({
          loading: true,
        })
        const postData = {}
        const keys = Object.keys(objValues)
        const values = Object.values(objValues)
        values.forEach((value, index) => (postData[keys[index]] = trim(value)))
        postData.roleIds = [postData.roleIds]

        this.props.handleSubmit(postData, () => {
          this.setState({
            loading: false,
          })
          this.props.toggleDialog()
          this.props.form.setFieldsValue({
            userId: '',
            userName: '',
            password: '',
            roleIds: '',
          })
          // 选中的数据清空
        }, () => {
          this.setState({
            loading: false,
          })
          showMessage({ text: 'Operation failed!', type: 'error' })
        })
      }
    })
  }

  render() {
    const { intl } = this.props
    const { getFieldDecorator } = this.props.form
    const formItemLayout = {
      labelCol: {
        xs: { span: 24 },
        sm: { span: 6 },
      },
      wrapperCol: {
        xs: { span: 24 },
        sm: { span: 18 },
      },
    }
    return (
      <Modal
        visible={this.props.isOpen}
        title={this.props.dialogTitle}
        onCancel={this.props.toggleDialog}
        footer={[
          <Button key="cancel" onClick={this.props.toggleDialog}>Cancel</Button>,
          <Button key="submit" type="primary" loading={this.state.loading} onClick={this.handleChange}>
            <WrapperFormattedMessage id='Save' />
          </Button>,
        ]}
      >
        <div style={{ padding: '0 20px 0 20px' }}>
          <Form>
            <FormItem
              {...formItemLayout}
              label={intl.formatMessage({ id: 'userAndLicenseManagement.license.userID' })}
            >
              {getFieldDecorator('userId', {
                rules: [{ required: true, message: intl.formatMessage({ id: 'userAndLicenseManagement.edit.userIDError' }) }],
              })(
                <Input disabled={this.props.dialogBtnContent.toLowerCase() === 'update'} type="text" placeholder={intl.formatMessage({ id: 'userAndLicenseManagement.license.userID' })} />,
              )}
            </FormItem>
            <FormItem
              {...formItemLayout}
              label={intl.formatMessage({ id: 'userAndLicenseManagement.license.userName' })}
            >
              {getFieldDecorator('userName', {
                rules: [{ required: true, message: intl.formatMessage({ id: 'userAndLicenseManagement.edit.userNameError' }) }],
              })(
                <Input type="text" placeholder="User Name" />,
              )}
            </FormItem>
            <FormItem
              {...formItemLayout}
              label={this.props.dialogBtnContent !== 'Update' ? 'Password' : <span style={{ paddingLeft: '10px' }}><WrapperFormattedMessage id='password' /></span>}
            >
              {getFieldDecorator('password', {
                rules: [{ required: this.props.dialogBtnContent !== 'Update', message: intl.formatMessage({ id: 'userAndLicenseManagement.edit.passwordError' }) }],
              })(
                <Input
                  suffix={<Icon
                    type="eye"
                    onMouseDown={this.mouseDownEye}
                    onMouseUp={this.mouseUpEye}
                  />}
                  type={this.state.passwordInputType}
                  placeholder={intl.formatMessage({ id: 'password' })}
                />,
              )}
            </FormItem>
            <FormItem
              {...formItemLayout}
              label={intl.formatMessage({ id: 'role' })}
            >
              {getFieldDecorator('roleIds', {
                rules: [{ required: true, message: intl.formatMessage({ id: 'userAndLicenseManagement.edit.roleError' }) }],
              })(
                <Select >
                  {(() => {
                    const roles = getRole()
                    const keys = Object.keys(roles)
                    const values = Object.values(roles)
                    return values.map((name, i) => (<Option
                      key={keys[i]}
                      value={keys[i]}
                    ><WrapperFormattedMessage id={name} /></Option>))
                  })()}
                </Select>
              )}
            </FormItem>

          </Form>
        </div>
      </Modal>
    )
  }
}


DialogFilter.propTypes = {
  dialogTitle: PropTypes.string.isRequired,
  dialogBtnContent: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  toggleDialog: PropTypes.func.isRequired,
  form: PropTypes.object.isRequired,
}

export default Form.create()(injectIntl(DialogFilter))
