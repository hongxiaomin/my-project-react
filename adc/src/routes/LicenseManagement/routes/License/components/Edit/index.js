import CreateUser from './Create'
import EditUser from './Edit'

export default { CreateUser, EditUser }
