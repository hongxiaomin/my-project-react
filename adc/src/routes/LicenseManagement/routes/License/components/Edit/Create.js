import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl } from 'react-intl'
import { connect } from 'react-redux'
import Dialog from './Dialog'
import { createUser } from '../../modules'


export class CreateUser extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.state = {
      isOpen: false,
    }
  }

  handleSubmit(values, cb, error) {
    this.props.dispatch(createUser(values, cb, error))
  }

  toggleDialog = () => {
    this.setState({
      isOpen: !this.state.isOpen,
    })
  }

  render() {
    const buttonText = {
      create: 'Create',
    }
    const titleText = {
      create: this.props.intl.formatMessage({ id: 'userAndLicenseManagement.edit.createUser' }),
    }
    const { type } = this.props
    return (
      <div>
        <button
          style={{ width: '80px' }}
          type='button'
          onClick={this.toggleDialog}
          className={'pt-button pt-intent-primary'}
        >
          {buttonText[this.props.type]}
        </button>
        <Dialog
          dialogTitle={titleText[type]}
          dialogBtnContent={buttonText[type]}
          isOpen={this.state.isOpen}
          handleSubmit={this.handleSubmit}
          toggleDialog={this.toggleDialog}
        />
      </div>
    )
  }
}

CreateUser.propTypes = {
  intl: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
}

export default connect(({ intl: { locale } }) => ({
  locale,
}))(injectIntl(CreateUser))
