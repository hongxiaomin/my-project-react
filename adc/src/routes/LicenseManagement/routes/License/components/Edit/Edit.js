import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl } from 'react-intl'
import { connect } from 'react-redux'
import { isEmpty } from 'lodash'
import Dialog from './Dialog'
import { updateUser } from '../../modules'

export class EditUser extends React.Component {

  componentWillReceiveProps(nextProps) {
    if (!isEmpty(nextProps.defaultValue)) {
      this.form.setFieldsValue({
        userId: nextProps.defaultValue.userId,
        userName: nextProps.defaultValue.userName,
        roleIds: nextProps.defaultValue.roleIds[0],
      })
    }
  }

  handleUpdate = (values, success, error) => {
    this.props.dispatch(updateUser(values, success, error))
  }

  render() {
    return (<Dialog
      ref={f => (this.form = f)}
      isOpen={this.props.isOpen}
      dialogTitle={this.props.intl.formatMessage({ id: 'userAndLicenseManagement.edit.EditUser' })}
      dialogBtnContent='Update'
      defaultValue={{}}
      handleSubmit={this.handleUpdate}
      toggleDialog={this.props.toggleDialog}
    />)
  }
}

EditUser.propTypes = {
  intl: PropTypes.object.isRequired,
  isOpen: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  toggleDialog: PropTypes.func.isRequired,
  defaultValue: PropTypes.object.isRequired,
}

export default connect()(injectIntl(EditUser))
