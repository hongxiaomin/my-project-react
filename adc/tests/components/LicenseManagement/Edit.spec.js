import React from 'react'
import { shallow, mount } from 'enzyme'
import { EditUser } from 'routes/LicenseManagement/routes/License/components/Edit/Edit'

describe('(Component) EditUser', () => {
  const isOpen = false
  const dispatch = jest.fn()
  const toggleDialog = jest.fn()
  const defaultValue = {}
  it('render normally', () => {
    const wrapper = mount(
      <EditUser
        isOpen={isOpen}
        dispatch={dispatch}
        toggleDialog={toggleDialog}
        defaultValue={defaultValue}
      />,
    )
    // expect(wrapper.ref('f').prop('dialogTitle')).to.equal('Edit User')
    expect(wrapper).toMatchSnapshot()
  })
//   it('render button click', () => {
//     const wrapper = shallow(
//       <EditUser
//         isOpen={isOpen}
//         dispatch={dispatch}
//         toggleDialog={toggleDialog}
//         defaultValue={defaultValue}
//       />,
//     )
//     wrapper.find('button').at(0).simulate('click')
//     expect(wrapper.state('isOpen')).toEqual(true)
//     wrapper.find('Dialog').at(0).simulate('change')
//     // expect(wrapper.state('isOpen')).toEqual(true)
//     expect(wrapper).toMatchSnapshot()
//   })
})
