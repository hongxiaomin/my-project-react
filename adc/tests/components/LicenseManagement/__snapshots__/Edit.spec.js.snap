// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) EditUser render normally 1`] = `
<EditUser
  defaultValue={Object {}}
  dispatch={[Function]}
  isOpen={false}
  toggleDialog={[Function]}
>
  <Form(WrappedComponent)
    defaultValue={Object {}}
    dialogBtnContent="Update"
    dialogTitle="Edit User"
    handleSubmit={[Function]}
    isOpen={false}
    toggleDialog={[Function]}
  >
    <Component
      defaultValue={Object {}}
      dialogBtnContent="Update"
      dialogTitle="Edit User"
      form={
        Object {
          "getFieldDecorator": [Function],
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldProps": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "isSubmitting": [Function],
          "resetFields": [Function],
          "setFields": [Function],
          "setFieldsInitialValue": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
          "validateFieldsAndScroll": [Function],
        }
      }
      handleSubmit={[Function]}
      isOpen={false}
      toggleDialog={[Function]}
      wrappedComponentRef={[Function]}
    >
      <DialogFilter
        defaultValue={Object {}}
        dialogBtnContent="Update"
        dialogTitle="Edit User"
        form={
          Object {
            "getFieldDecorator": [Function],
            "getFieldError": [Function],
            "getFieldInstance": [Function],
            "getFieldProps": [Function],
            "getFieldValue": [Function],
            "getFieldsError": [Function],
            "getFieldsValue": [Function],
            "isFieldTouched": [Function],
            "isFieldValidating": [Function],
            "isFieldsTouched": [Function],
            "isFieldsValidating": [Function],
            "isSubmitting": [Function],
            "resetFields": [Function],
            "setFields": [Function],
            "setFieldsInitialValue": [Function],
            "setFieldsValue": [Function],
            "submit": [Function],
            "validateFields": [Function],
            "validateFieldsAndScroll": [Function],
          }
        }
        handleSubmit={[Function]}
        isOpen={false}
        toggleDialog={[Function]}
        wrappedComponentRef={[Function]}
      >
        <Modal
          confirmLoading={false}
          footer={
            Array [
              <Button
                clicked={false}
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
            >
                Cancel
            </Button>,
              <Button
                clicked={false}
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
                type="primary"
            >
                Save
            </Button>,
            ]
          }
          maskTransitionName="fade"
          onCancel={[Function]}
          prefixCls="ant-modal"
          title="Edit User"
          transitionName="zoom"
          visible={false}
          width={520}
        >
          <DialogWrap
            confirmLoading={false}
            footer={
              Array [
                <Button
                  clicked={false}
                  ghost={false}
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
              >
                  Cancel
              </Button>,
                <Button
                  clicked={false}
                  ghost={false}
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                  type="primary"
              >
                  Save
              </Button>,
              ]
            }
            maskTransitionName="fade"
            onCancel={[Function]}
            onClose={[Function]}
            prefixCls="ant-modal"
            title="Edit User"
            transitionName="zoom"
            visible={false}
            width={520}
          />
        </Modal>
      </DialogFilter>
    </Component>
  </Form(WrappedComponent)>
</EditUser>
`;
